### PR TITLE
Set up eslint for vue/typescript

### PR DESCRIPTION
**Pull request by Silvan-WMDE**
This sets up JavaScript/TypeScript linting via eslint and adds a corresponding GitHub action.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 - [Working with OAuth](#working-with-oauth)
 - [Frontend - Working with CSS and JS](#frontend---working-with-css-and-js)
 - [PHP Linting](#php-linting)
+- [Javascript Linting](#js-linting)
```
Comment (itamargiv) [inline docs/README.md:15]: Cool thing, thanks for adding the docs! Do you think these two sections should perhaps live under the same higher level heading "Linting", like such (see below)?

```markdown
## Linting

### PHP Linting

<!-- content... -->

### Javascript Linting

<!-- content... -->
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
     "private": true,
     "scripts": {
+        "lint": "eslint --ext .js,.vue resources/js",
+        "lint:fix": "eslint --fix --ext .js,.vue resources/js",
```
Comment (itamargiv) [inline package.json:5]: Nice! :+1: Thank you

**Review by itamargiv — COMMENTED** [review]
Looks great! I left one comment for discussion, but not sure if it's a blocker :100:

File: docs/README.md (hunk 1 above)
Comment (Silvan-WMDE) [reply docs/README.md]: Yes, that makes sense. Thank you.

**Review by itamargiv — APPROVED** [review]
Thank you!

**Pull request merged**